transport/src/main/java/io/netty/bootstrap/Bootstrap.java

```diff
@@ -194,44 +194,47 @@ public void operationComplete(ChannelFuture future) throws Exception {
-        final AddressResolver<SocketAddress> resolver = this.resolver.getResolver(eventLoop);
-
-        if (!resolver.isSupported(remoteAddress) || resolver.isResolved(remoteAddress)) {
-            // Resolver has no idea about what to do with the specified remote address or it's resolved already.
-            doConnect(remoteAddress, localAddress, promise);
-            return promise;
-        }
-
-        final Future<SocketAddress> resolveFuture = resolver.resolve(remoteAddress);
+        try {
+            final EventLoop eventLoop = channel.eventLoop();
+            final AddressResolver<SocketAddress> resolver = this.resolver.getResolver(eventLoop);
+
+            if (!resolver.isSupported(remoteAddress) || resolver.isResolved(remoteAddress)) {
+                // Resolver has no idea about what to do with the specified remote address or it's resolved already.
+                doConnect(remoteAddress, localAddress, promise);
+                return promise;
+            }
 
-        if (resolveFuture.isDone()) {
-            final Throwable resolveFailureCause = resolveFuture.cause();
+            final Future<SocketAddress> resolveFuture = resolver.resolve(remoteAddress);
 
-            if (resolveFailureCause != null) {
-                // Failed to resolve immediately
-                channel.close();
-                promise.setFailure(resolveFailureCause);
-            } else {
-                // Succeeded to resolve immediately; cached? (or did a blocking lookup)
-                doConnect(resolveFuture.getNow(), localAddress, promise);
-            }
-            return promise;
-        }
+            if (resolveFuture.isDone()) {
+                final Throwable resolveFailureCause = resolveFuture.cause();
 
-        // Wait until the name resolution is finished.
-        resolveFuture.addListener(new FutureListener<SocketAddress>() {
-            @Override
-            public void operationComplete(Future<SocketAddress> future) throws Exception {
-                if (future.cause() != null) {
+                if (resolveFailureCause != null) {
+                    // Failed to resolve immediately
                     channel.close();
-                    promise.setFailure(future.cause());
+                    promise.setFailure(resolveFailureCause);
                 } else {
-                    doConnect(future.getNow(), localAddress, promise);
+                    // Succeeded to resolve immediately; cached? (or did a blocking lookup)
+                    doConnect(resolveFuture.getNow(), localAddress, promise);
                 }
+                return promise;
             }
-        });
 
+            // Wait until the name resolution is finished.
+            resolveFuture.addListener(new FutureListener<SocketAddress>() {
+                @Override
+                public void operationComplete(Future<SocketAddress> future) throws Exception {
+                    if (future.cause() != null) {
```

Comment:
@normanmaurer nit: save `cause()` in local variable to avoid 2 volatile reads (or atomic reads)?